src/test/ui/impl-trait/issues/issue-86201.stderr

```diff
@@ -7,7 +7,7 @@ LL |
 LL | static STATIC_FN: FunType = some_fn;
    |                             ^^^^^^^ expected opaque type, found fn item
    |
-   = note: expected opaque type `impl Fn<()>`
+   = note: expected opaque type `impl Fn<()> + FnOnce<()>`
```

Comment:
Noticed this, a bit odd because it seems you were detecting this case already

Reply:
So the reason this one doesn't rewrite itself into `impl Fn() -> T` is because the opaque type has no return type constraint. I'll fix it to just go back to `impl Fn<()>` (very easy to do) or we could print `-> _` if you think that would look better!

Reply:
Either approach would be fine. Do whichever works best for you.